src/lib/src/contents-dialog/tableOfContents/tableOfContents.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+
+import { MimeViewerIntl } from '../../core/intl/viewer-intl';
+import { IiifManifestService } from '../../core/iiif-manifest-service/iiif-manifest-service';
+import { Canvas, Manifest, Structure } from '../../core/models/manifest';
+import { ViewerService } from '../../core/viewer-service/viewer.service';
+import { ContentsDialogComponent } from '../contents-dialog.component';
+import { MdDialogRef } from '@angular/material';
+import { ObservableMedia } from '@angular/flex-layout';
+import { PageService } from '../../core/page-service/page-service';
+
+@Component({
+  selector: 'mime-toc',
+  templateUrl: './tableOfContents.component.html',
+  styleUrls: ['./tableOfContents.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TOCComponent implements OnInit, OnDestroy {
```

Comment:
File and directory name should be the same as class name

TocComponent should be toc.component

Use TocComponent instead of TOCComponent